packages/mui-system/src/borders.js

```diff
@@ -67,20 +67,26 @@ export const borderLeftColor = style({
   themeKey: 'palette',
 });
 
-// false positive
-// eslint-disable-next-line react/function-component-definition
-export const borderRadius = (props) => {
-  if (props.borderRadius !== undefined && props.borderRadius !== null) {
-    const transformer = createUnaryUnit(props.theme, 'shape.borderRadius', 4, 'borderRadius');
-    const styleFromPropValue = (propValue) => ({
-      borderRadius: getValue(transformer, propValue),
-    });
-    return handleBreakpoints(props, props.borderRadius, styleFromPropValue);
-  }
-
-  return null;
+/**
+ *
+ * @param {'borderRadius' | 'borderTopLeftRadius' | 'borderTopRightRadius' | 'borderBottomLeftRadius' | 'borderBottomRightRadius'} cssKey
+ */
+export const createBorderRadiusStyle = (cssKey) => {
+  return (props) => {
+    if (props[cssKey] !== undefined && props[cssKey] !== null) {
+      const transformer = createUnaryUnit(props.theme, 'shape.borderRadius', 4, cssKey);
+      const styleFromPropValue = (propValue) => ({
+        [cssKey]: getValue(transformer, propValue),
+      });
+      return handleBreakpoints(props, props[cssKey], styleFromPropValue);
+    }
+
+    return null;
+  };
 };
 
+export const borderRadius = createBorderRadiusStyle('borderRadius');
+
```

Comment:
I think we should use `createBorderRadiusStyle('borderRadius')` for `borderRadius` in `defaultSxConfig.js` the same way it's done for the other properties (`top-left`, `top-right`, ...) instead of here.

Reply:
This is for backward compatibility. Since `borderRadius` was already exported previously, I have created borderRadius here only. For others, I have directly called the function in `defaultSxConfig` since they are new.

Reply:
Is `borderRadius` exported to users? Or just for internal use?

Reply:
Most of the first level directories/files are already available to users. So `borderRadius` can be imported by

```js
import {borderRadius} from '@mui/system';
// or
import {borderRadius} from '@mui/system/borders'
```

Reply:
I see. Yeah let's keep it backward compatible 👍🏼